### PR TITLE
[flang]: This is to fix the HLFIR path for PPC Vector type intrinsics.

### LINKED ISSUE #65928: [flang][hlfir] PowerPC vector types needs support in HLFIR
PowerPC vector types needs some support in HLFIR lowering, compiling code from test/Lower/PowerPC with the -flang-experimental-hlfir option currently crashes in Lowering.

At least the following is needed to add support:

- The fir.vector type must likely be allowed in the `isFortranValueType` [predicate here](https://github.com/llvm/llvm-project/blob/be5b66670d869f5f9401feb71c70f2ca519ce9e0/flang/include/flang/Optimizer/Builder/HLFIRTools.h#L47C13-L47C31).
- Special handling should be added in assignment hlfir lowering code [here](https://github.com/llvm/llvm-project/blob/be5b66670d869f5f9401feb71c70f2ca519ce9e0/flang/lib/Lower/Bridge.cpp#L3352).

### PR DESCRIPTION
**Pull request by DanielCChen**
PowerPC Vector type intrinsics currently crashes with `-flang-experimental-hlfir` is specified.

This patch is to fix the HLFIR path for PowerPC Vector type intrinsics.

### CODE REVIEW

**Review by jeanPerier — COMMENTED** [review]
Thank you very much for the quick fix @DanielCChen!
Do all the PPC tests produce correct IR with the -hlfir option with this patch? If so, could you update the relevant ones here with a new RUN line with bbc -emit-fir -hlfir to avoid regressions (if the tested output is the same, you can probably use the same check-prefix as without the -hlfir option)?

File: flang/include/flang/Optimizer/Dialect/FIRType.h [hunk 1]
```diff
@@ -165,7 +170,7 @@ inline bool isa_char(mlir::Type t) { return t.isa<fir::CharacterType>(); }
 /// Is `t` a trivial intrinsic type? CHARACTER is <em>excluded</em> because it
 /// is a dependent type.
 inline bool isa_trivial(mlir::Type t) {
-  return isa_integer(t) || isa_real(t) || isa_complex(t) ||
+  return isa_integer(t) || isa_real(t) || isa_complex(t) || isa_vector(t) ||
```
Comment (tblah) [inline flang/include/flang/Optimizer/Dialect/FIRType.h:173]: Why is a vector trivial? Vectors can be containers for other types and could be very large in memory

**Comment by DanielCChen** [comment]
> Thank you very much for the quick fix @DanielCChen! Do all the PPC tests produce correct IR with the -hlfir option with this patch? If so, could you update the relevant ones here with a new RUN line with bbc -emit-fir -hlfir to avoid regressions (if the tested output is the same, you can probably use the same check-prefix as without the -hlfir option)?

Hi Jean, I tried a couple of test cases and it seems both FIR and LLVM IR are the same as before. We use both `bbc` (to generate FIR) and `flang_fc1` (to generate llvm IR). I will update all of the test cases and run again.

**Comment by jeanPerier** [comment]
Thanks for the update @DanielCChen. Given this is target specific, it makes sense to me to only keep the LLVM IR lit tests to reduce maintenance.

**Comment by DanielCChen** [comment]
After further investigation, the differences in LLVM IR on the HLFIR path is non-functional. The assembly code are the same and it produces the correct result. 
I have updated the LLVM IR with the 3rd commit in this PR, and verified all PowerPC vector intrinsic testing is passing with HLFIR path.
@jeanPerier

**Review by jeanPerier — APPROVED** [review]
Thank you very much!

I am not sure why the format checking is not able to check your patch (fails with a weird github action error "Unable to find merge base" while github claims that "Merging can be performed automatically" on your PR).

Maybe a rebase would solve that, but this may break the comment threads, so I am fine if you just merge, I manually verified your formatting is good with git-clang-format (I am also OK with a rebase and force push into this PR to try a fresh build bot check, do whatever you are the most comfortable with).

**Comment by DanielCChen** [comment]
> Thank you very much!
> 
> I am not sure why the format checking is not able to check your patch (fails with a weird github action error "Unable to find merge base" while github claims that "Merging can be performed automatically" on your PR).
> 
> Maybe a rebase would solve that, but this may break the comment threads, so I am fine if you just merge, I manually verified your formatting is good with git-clang-format (I am also OK with a rebase and force push into this PR to try a fresh build bot check, do whatever you are the most comfortable with).

Thanks @jeanPerier ! I synch the forked repo and rebased the PR branch. It seems worked!

**Pull request merged**